### PR TITLE
Google natural lang sentiment

### PR DESCRIPTION
**Pull request by dantan380**

### CODE REVIEW

**Comment by garrickhenne** [comment]
I think we’re trying to merge into features/create-review instead of main

File: backend/apis/google-natural-lang-api.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  };
+
+  interface IDocument {
+    content: string;
+    type: "PLAIN_TEXT" | "HTML" | "TYPE_UNSPECIFIED"; // Ensure it matches the expected values/types
+    // Other properties...
+  }
+
+  // Detects sentiment of entities in the document
+  const [result] = await languageClient.analyzeEntitySentiment({ document });
```
Comment (robertshum) [inline backend/apis/google-natural-lang-api.ts:27]: can we surround the await statement in a try catch block?

I think this (the code you have below) handles uncaught errors thrown, but it lives in the process so it's global.  Kinda weird to catch all exceptions in the application that lives in a google API file.

```js
process.on('unhandledRejection', err => {
  console.error(err);
  process.exitCode = 1;
});
```

**Comment by robertshum** [comment]
hmm merge conflicts in package-lock.json, I wonder if we can just delete the local package-lock.json and then rebuild it with npm install.

**Review by robertshum — COMMENTED** [review]
just the comment about the try catch block and we should be able to remove the process.on code that listens for all errors.

**Comment by dantan380** [comment]
I used '.then' and '.catch' instead, would this be okay?
`languageClient.analyzeEntitySentiment({ document })
    .then(([result]) => {
      const entities: any = result.entities;
      calculateSentimentAverage(entities);
    })
    .catch((error) => {
      console.error('Error:', error);
    });`

**Review by robertshum — APPROVED** [review]
Looks good!

**Review by garrickhenne — APPROVED** [review]

**Pull request merged**